From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aswin-mcw <aswin.kalies@multicorewareinc.com>
Date: Wed, 9 Sep 2026 13:14:47 +0530
Subject: [PATCH 065/109] Add conditionals for AArch64 and update MinGW
 program checks to use build_cpu

(cherry picked from commit 7b67660e300b17ee7f5a009c1bf011f239112d8b)
---
 winsup/configure.ac | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/winsup/configure.ac b/winsup/configure.ac
index 923da96fd..4702ba849 100644
--- a/winsup/configure.ac
+++ b/winsup/configure.ac
@@ -109,6 +109,8 @@ esac
 
 AC_SUBST(DLL_ENTRY)
 
+AM_CONDITIONAL(BUILD_X86_64, [test $build_cpu = "x86_64"])
+AM_CONDITIONAL(BUILD_AARCH64, [test $build_cpu = "aarch64"])
 AM_CONDITIONAL(TARGET_X86_64, [test $target_cpu = "x86_64"])
 AM_CONDITIONAL(TARGET_AARCH64, [test $target_cpu = "aarch64"])
 
@@ -148,10 +150,10 @@ if test -z "$XMLTO"; then
     fi
 fi
 
-if test "x$with_cross_bootstrap" != "xyes"; then
-    AC_CHECK_PROGS(MINGW_CXX, ${target_cpu}-w64-mingw32-g++)
+if test "x$with_mingw_progs" != "xno"; then
+    AC_CHECK_PROGS(MINGW_CXX, ${build_cpu}-w64-mingw32-g++)
     test -n "$MINGW_CXX" || AC_MSG_ERROR([no acceptable MinGW g++ found in \$PATH])
-    AC_CHECK_PROGS(MINGW_CC, ${target_cpu}-w64-mingw32-gcc)
+    AC_CHECK_PROGS(MINGW_CC, ${build_cpu}-w64-mingw32-gcc)
     test -n "$MINGW_CC" || AC_MSG_ERROR([no acceptable MinGW gcc found in \$PATH])
 
     AC_CHECK_PROGS(MINGW32_CC, i686-w64-mingw32-gcc)
